src/client/src/Stream.c

```diff
@@ -900,7 +900,7 @@ STATUS putFrame(PKinesisVideoStream pKinesisVideoStream, PFrame pFrame)
         CHK_STATUS(mkvgenPackageFrame(pKinesisVideoStream->pMkvGenerator, pFrame, pTrackInfo, pAlloc, &packagedSize, &encodedFrameInfo));
 
         // Package the metadata if specified
-        if (packagedMetadataSize != 0) {
+        if (packagedMetadataSize != 0 && !justStartedStreaming) {
```

Comment:
`justStartedStreaming` is never reset so it will always be `FALSE`

Reply:
Also we should now add a test which actually tests if we block tags before first cluster which is the desired outcome after this change.